src/main/java/training/filereading/JSONRecordReader.java

```diff
@@ -0,0 +1,66 @@
+package training.filereading;
+
+import com.google.gson.*;
+import com.google.gson.reflect.TypeToken;
+import training.bank.TransactionRecord;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.math.BigDecimal;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.HashSet;
+import java.util.List;
+
+public class JSONRecordReader implements Reader {
```

Comment:
Let me show you a better way of doing this later on. Come and grab me at some point in the morning.